### PR TITLE
lessons 2-13 working

### PR DESCRIPTION
**Pull request by wwwtyro**

### CODE REVIEW

**Comment by wwwtyro** [comment]
@ibgreen not sure if you get pinged when I make a PR, so pinging you here in case you want to merge. :tada:

File: examples/lessons/11/index.js [hunk 1]
```diff
@@ -51,7 +50,8 @@ function webGLStart() {
         camera.update();
       }
     },
-    onError: function() {
```
Comment (ibgreen) [inline examples/lessons/11/index.js]: Isn't this the default implementation? remove from examples?

File: src/core.js [hunk 2]
```diff
@@ -66,11 +66,8 @@ const globalContext = typeof window !== 'undefined' ? window : global;
 
 // Creates a single application object asynchronously
 // with a gl context, a camera, a program, a scene, and an event system.
-export async function PhiloGL(canvasId, opt = {}) {
-  opt = {
-    ...DEFAULT_OPTS,
-    ...opt
-  };
+export function PhiloGL(canvasId, opt = {}) {
+  opt = $.merge(DEFAULT_OPTS, opt);
```
Comment (ibgreen) [inline src/core.js]: Use spread instead of $.merge. Would like to gradually phase out jquery-mini.

File: src/event.js [hunk 3]
```diff
@@ -404,13 +404,13 @@ export const Events = {
     };
 
     var bind = opt.bind;
-
     if (bind) {
-      for (var name in opt) {
+      for (let name in opt) {
```
Comment (ibgreen) [inline src/event.js]: const name in opt should be better.

Try 'npm run lint' before you commit, it should warn about this.

Also if you are using sublime you can install sublime-linter and it should highlight the linter errors inline in the code.

File: src/program.js [hunk 4]
```diff
@@ -295,13 +314,13 @@ export default class Program {
   // Build program from default shaders (requires Shaders)
   static fromDefaultShaders(opt = {}) {
     const {vs = 'Default', fs = 'Default'} = opt;
-    const sh = PhiloGL.Shaders;
+    const sh = Shaders;
```
Comment (ibgreen) [inline src/program.js]: Let's type out the word shaders.

**Comment by ibgreen** [comment]
Thanks Rye, great to see the progress! Note that some of my comments are on the previous (closed) pull request, and some on this.

File: src/core.js (hunk 2 above)
Comment (wwwtyro) [inline src/core.js]: @ibgreen The spread operator only performs a shallow merge. How about lodash.defaultsDeep?

**Comment by wwwtyro** [comment]
@ibgreen Thanks for another thorough review! I'll make these changes first thing tonight.

File: examples/lessons/11/index.js (hunk 1 above)
Comment (wwwtyro) [inline examples/lessons/11/index.js]: The default defined in core.js doesn't have the alert.

**Comment by wwwtyro** [comment]
@ibgreen Alright, all lessons working! I'll start a new branch based on this one for the examples.

**Pull request merged**